core/runtime/jsruntime.rs

```diff
@@ -375,8 +375,10 @@ pub type CompiledWasmModuleStore = CrossIsolateStore<v8::CompiledWasmModule>;
 /// Internal state for JsRuntime which is stored in one of v8::Isolate's
 /// embedder slots.
 pub struct JsRuntimeState {
-  pub(crate) source_map_getter: Option<Rc<Box<dyn SourceMapGetter>>>,
-  pub(crate) source_map_cache: Rc<RefCell<SourceMapCache>>,
+  pub(crate) source_mapper: RefCell<SourceMapper<Box<dyn SourceMapGetter>>>,
+  // This is not the right place for this, but it's the easiest way to make
+  // op_apply_source_map a fast op. This stashing should happen in #[op2].
+  pub(crate) stashed_file_name: RefCell<Option<String>>,
```

Comment:
Can't we move it to `SourceMapper`?

Reply:
I think it makes more sense in the runtime state rather than the source mapper. That is because the source mapper's job is to provide and cache source maps. A more appropriate place for it could be in OpState or as an element of the GothamState.

Reply:
I'm not sure I agree - previously this field lived on the `SourceMapCache` and it's limited in use for source mapping. It feels quite natural to "hide it" in `SourceMapper`